tensorflow_recommenders/layers/factorized_top_k_test.py

```diff
@@ -166,38 +180,50 @@ def test_scann(self, identifier_dtype):
     scann = factorized_top_k.ScaNN()
     scann.index(candidates, candidate_names)
 
-    for _ in range(100):
-      pre_serialization_results = scann(query[:2])
-
-    path = os.path.join(self.get_temp_dir(), "query_model")
-    scann.save(
-        path,
-        options=tf.saved_model.SaveOptions(namespace_whitelist=["Scann"]))
-    loaded = tf.keras.models.load_model(path)
-
-    for _ in range(100):
-      post_serialization_results = loaded(tf.constant(query[:2]))
-
-    self.assertAllEqual(post_serialization_results, pre_serialization_results)
+    self.run_save_and_restore_test(scann, query, 100)
 
   def test_scann_dataset_arg_no_identifiers(self):
 
-    num_candidates = 100
+    num_candidates, num_queries = (100, 4)
+
+    rng = np.random.RandomState(42)
     candidates = tf.data.Dataset.from_tensor_slices(
-        np.random.normal(size=(num_candidates, 4)).astype(np.float32))
+        rng.normal(size=(num_candidates, 4)).astype(np.float32))
+    query = rng.normal(size=(num_queries, 4)).astype(np.float32)
 
-    index = factorized_top_k.ScaNN()
-    index.index(candidates.batch(100))
+    scann = factorized_top_k.ScaNN()
+    scann.index(candidates.batch(100))
+
+    self.run_save_and_restore_test(scann, query, 100)
```

Comment:
In `test_scann_dataest_arg_no_identifiers`, `num_reordering_candidates` is `None`.
For cases where `num_reordering_candidates` is not `None`, it is tested in `test_scann_top_k`